File: src/SortAlgorthm/insertionsort/insertionSort.java
package SortAlgorthm.insertionsort;

public class insertionSort {


    public static int[]insertionSort(int[]array){

           for (int i = 1 ; i<array.length;i++){
               int  data = array[i] ;
               int j = i ;


               while (j > 0 && data< array[j-1]) {
                   int value = array[j];

                   //swap

                   array[j] = array[j-1];
                   array[j-1]=value ;
                   j--;

               }

           }
       return array  ;
    }


    public static void main(String args[])
    {

        int []arr = {4,7,1,5,3,2,6} ;
        int[] ints = insertionSort(arr);
        for (int i : ints){

            System.out.println(i);
        }


    }
}

File: src/SortAlgorthm/insertionsort/Insertion_Sort.kt
package sortAndsearch_Algorthm.insertionsort

class Insertion_Sort {


    fun insertionSort(arr:IntArray):IntArray{

         for (i in 1 until arr.size){

             val element =arr[i]
             var j=i-1//2
             while (j>=0&& arr[j]>element){//2

                 arr[j+1]=arr[j]
                 j--//1
             }
             arr[j+1] = element
         }

        return arr
    }

    fun Insertion_Sort_2(arr:IntArray):IntArray{

        var key:Int
        var j = 0

        for (i in 1..arr.size-1){
            key=arr[i]
            j=i-1
            while (j>=0&& arr[j]>key){
                arr[j+1]=arr[j]
                j--

            }
            arr[j+1]= key
        }


        return arr
    }
}